data/mp/multiplay/skirmish/bonecrusher/produce.js

```diff
@@ -0,0 +1,259 @@
+//Подготавливаем технологии для производства
+//на самом деле лишняя функция, однако необходимая
+//для того, что бы не читерить. Мы не можем производить
+//войска по исследованным технологиям без HQ, так этой функцией запретим
+//это делать и ИИ
+function prepeareProduce(){
+//	debugMsg('prepeareProduce()', 'production')
+	//Если есть HQ
+	var hq = enumStruct(me, HQ).filter(function (e){if(e.status == BUILT)return true;return false;});
+	if (hq.length != 0){
+		
+		
+		//Составляем корпуса
+		light_bodies=[];
+		medium_bodies=[];
+		heavy_bodies=[];
+		bodies.forEach(function(e){
+			//			debugMsg("Body: "+e[1]+" "+getResearch(e[0]).done, 'production');
+			switch (e[1]){
+				case "Body1REC":
+				case "Body4ABT":
+				case "Body2SUP":
+				case "Body3MBT":
+					if(getResearch(e[0]).done) light_bodies.unshift(e[1]);
+					   break;
+				case "Body5REC":
+				case "Body8MBT":
+				case "Body6SUPP":
+				case "Body7ABT":
+					if(getResearch(e[0]).done) medium_bodies.unshift(e[1]);
+					   break;
+				case "Body11ABT":
+				case "Body12SUP":
+				case "Body9REC":
+				case "Body10MBT":
+				case "Body13SUP":
+				case "Body14SUP":
+					if(getResearch(e[0]).done) heavy_bodies.unshift(e[1]);
+					   break;
+					
+			}
+		});
+		
+		
+		/*
+		//Сортируем пушки по "крутизне", базируясь на research.points
+		var _guns=guns.filter(function(e){
+			debugMsg(e[0]+' - '+getResearch(e[0]).done, 'weap');
+			if(getResearch(e[0]).done)return true;return false;
+		}).sort(function (a,b){
+			if(getResearch(a[0]).points < getResearch(b[0]).points ) return -1;
+			if(getResearch(a[0]).points > getResearch(b[0]).points ) return 1;
+			return 0;
+		});
+		
+		avail_guns=[];
+		for (var i in _guns){
+			avail_guns.push(_guns[i][1]);
+//			debugMsg(getResearch(_guns[i][0]).points+" "+_guns[i][0]+"->"+_guns[i][1], 'weap');
+			debugMsg(getResearch(_guns[i][0]).points+" - "+research_name[_guns[i][0]], 'weap');
+		}
+		if(avail_guns.length > 2) avail_guns.shift(); //Выкидываем первый пулемётик
+		if(avail_guns.length > 2) avail_guns.shift(); //Выкидываем спаренный пулемётик
+		avail_guns.reverse();
+		*/
+
+		//Дай мне три типа лучших пушек на данный момент
+		avail_guns = _weaponsGetGuns(3);
+		
+//		for(i in avail_guns) debugMsg(avail_guns[i], 'weap');
+		
+		//Сайборги заполонили!
+		var avail_cyborgs=[];
```

Comment:
This is a global, remove the `var`. Cyborg production is broken otherwise.

Reply:
Hmm.
I will run several tests before changes and commit this situation..